test/unittests/testutil.cc

```diff
@@ -245,7 +247,55 @@ MockCatalog* MockCatalog::AttachFreely(const std::string  &root_path,
   return new_catalog;
 }
 
-void MockCatalog::RegisterChild(MockCatalog *child) {
+void MockCatalog::RemoveChild(MockCatalog *child) {
+  std::vector<NestedCatalog>::iterator iter = active_children_.begin();
+  bool found = false;
+  while (!found && iter != active_children_.end()) {
+    if (iter->hash == child->hash()) {
+      active_children_.erase(iter);
+      found = true;
+    }
+    iter++;
+  }
+}
+
+MockCatalog* MockCatalog::FindSubtree(const PathString &path) const {
+  for (unsigned i = 0; i < active_children_.size(); ++i) {
+    if (active_children_[i].path == path)
+      return active_children_[i].child;
+  }
+  return NULL;
+}
+
+bool MockCatalog::LookupPath(const PathString &path,
+                catalog::DirectoryEntry *dirent) const {
+  shash::Md5 md5_path(path.GetChars(), path.GetLength());
+  for (unsigned i = 0; i < files_.size(); ++i) {
+    if (files_[i].path_hash == md5_path) {
+      *dirent = files_[i].ToDirectoryEntry();
+      return true;
+    }
+  }
+  return false;
+}
+
+bool MockCatalog::ListingPath(const PathString &path,
+                 catalog::DirectoryEntryList *listing) const {
+  listing->clear();
```

Comment:
No, it doesn't. It directly pushes back the new DirectoryEntries. However, I don't really see the point of not clearing the list before writing to it. Actually, when making a lookup the DirectoryEntry is at the beginning initialised to an empty value. The same way it could (and in my opinion should) be done in the CatalogManager, isn't it?